server/findproviders.go

```diff
@@ -1,98 +1,132 @@
 package server
 
 import (
-	"bytes"
-	"encoding/json"
+	"context"
 	"net/http"
 
 	"github.com/ipfs/go-cid"
 	"github.com/ipfs/go-delegated-routing/client"
-	"github.com/ipfs/go-delegated-routing/parser"
+	proto "github.com/ipfs/go-delegated-routing/gen/proto"
 	logging "github.com/ipfs/go-log"
+	"github.com/ipld/edelweiss/values"
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/multiformats/go-multiaddr"
 )
 
-var log = logging.Logger("delegated/server")
+var logger = logging.Logger("service/server/DelegatedRouting")
 
-type FindProvidersAsyncFunc func(cid.Cid, chan<- client.FindProvidersAsyncResult) error
+type DelegatedRoutingService interface {
```

Comment:
Identify is handled in the auto-generated code. Regarding client behavior:
I don't think that we should bake in an initial call to Identify by default. This would add a roundtrip to use cases where a client is used only once. Come to think of it, if we want to "cache" what methods are available, we should do it lazily: make an optimistic call to some method and then remember if it worked out or not (for future calls). That latter functionality does not belong here anyway. It is generic, so it should live in Edelweiss. What do you think?

Reply:
Do we have a way of knowing that a given method is unsupported rather than there being some intermittent issue?

Also, probably obvious but in the identify case we could save the round-trip by querying in parallel. It would add an extra query (particularly annoying in the repeated single-use case), but without the added latency.

Reply:
The error indicating a method is missing or the schema is unsupported is returned in the first asynchronous result. I can:
- make these errors distinguishable (from others)
- update the generated client code to cache the information that a method is missing (and not attempt further calls and return an error to the user right away)

This solution would remove the need for calling Identify. @aschmahmann  What do you think?

Reply:
Seems like a reasonable strategy, although I wonder if `Identify` is still a helpful thing to have around whether for debugging or as an alternative mechanism for figuring out what's supported.

If it seems unhelpful/wasteful we could go back to the spec and drop it given this is the first implementation. WDYT?

Reply:
I would keep it for now. I see a potential/eventual/useful extension of the Identify method's response which returns the schema (request and response) for each supported method.